extensions/objects/models/objects_test.py

```diff
@@ -900,3 +913,99 @@ def test_normalize(self):
         with self.assertRaisesRegexp(
             TypeError, 'Unexpected tab characters in code string: \t'):
             code_string.normalize('\t')
+
+
+class BaseTranslatableObjectTests(test_utils.GenericTestBase):
+
+    def test_translatable_objects_naming(self):
+        for name, member in inspect.getmembers(objects):
+            if not inspect.isclass(member):
+                continue
+
+            if isinstance(member(), objects.BaseTranslatableObject):
+                if name == 'BaseTranslatableObject':
+                    continue
+                self.assertEqual(name.index('Translatable'), 0)
+            else:
+                self.assertTrue('Translatable' not in name)
+
+    def test_translatable_required_methods(self):
+        mock_translatable_object_class = type(
```

Comment:
I'm a bit confused about this. Why are we constructing our own types and not just using existing ones?